.github/workflows/macos-generation.yml

```diff
@@ -0,0 +1,198 @@
+        -var="github_api_pat=${{ secrets.GH_FEED_TOKEN }}" `
+        -var="build_id=${{ inputs.virtualMachineName }}" `
+        -var="baseimage_name=${{ inputs.base_image_name }}" `
+        -var="xcode_install_user=$XcodeUser" `
+        -var="xcode_install_password=$XcodePassword" `
+        -color=false `
+        ${{ inputs.template_path }} `
+        | Where-Object {
+            #Filter sensitive data from Packer logs
+            $currentString = $_
+            $sensitiveString = $SensitiveData | Where-Object { $currentString -match $_ }
+            $sensitiveString -eq $null
+        }
+      working-directory: images/macos
+      env:
+        PACKER_LOG: 1
+        PACKER_LOG_PATH: ${{ runner.temp }}/packer-log.txt
+
+    - name: Prepare artifact
+      shell: bash
+      run: |
+        echo "Preparing artifact directory"
+        mkdir -p ${{ runner.temp }}/artifacts
+
+        echo "Copy image output files"
+        cp -R "images/image-output/software-report/." "${{ runner.temp }}/artifacts"
+
+        echo "Put VM name to 'VM_Done_Name' file"
+        echo "${{ inputs.virtualMachineName }}" > "${{ runner.temp }}/artifacts/VM_Done_Name"
+
+    - name: Print markdown software report
+      shell: bash
+      run: |
+        cat "${{ runner.temp }}/artifacts/systeminfo.md"
+
+    - name: Print json software report
+      shell: bash
+      run: |
+        cat "${{ runner.temp }}/artifacts/systeminfo.json"
+
+    - name: Publish Artifacts
+      uses: actions/upload-artifact@v3
+      with:
+        name: Built_VM_Artifacts
+        path: ${{ runner.temp }}/artifacts/
+
+    - name: Print provisioners duration
+      run: |
+        ./images.CI/measure-provisioners-duration.ps1 `
+          -PackerLogPath "${{ runner.temp }}/packer-log.txt" `
+          -PrintTopNLongest 25
+
+    - name: Move vm to cold storage and clear datastore tag
+      run: |
+        $VIUserName = az keyvault secret show -n "vcenter-username-v2" --vault-name "${{ env.KEYVAULT }}" --query value -o tsv
+        $VIPassword = az keyvault secret show -n "vcenter-password-v2" --vault-name "${{ env.KEYVAULT }}" --query value -o tsv
+        $cpuCount = 3
+        $coresPerSocketCount = 3
+        $memory = 14336
+
+        ./images.CI/macos/move-vm.ps1 `
+          -VMName "${{ inputs.virtualMachineName }}" `
+          -TargetDataStore "${{ inputs.target_datastore }}" `
+          -VIServer "${{ secrets.VISERVER_V2 }}" `
+          -VIUserName "$VIUserName" `
+          -VIPassword "$VIPassword" `
+          -CpuCount "$cpuCount" `
+          -CoresPerSocketCount "$coresPerSocketCount" `
+          -Memory "$memory"
+
+    - name: Destroy VM (if build canceled only)
+      if: ${{ cancelled() }}
+      run: |
+        $VIUserName = az keyvault secret show -n "vcenter-username-v2" --vault-name "${{ env.KEYVAULT }}" --query value -o tsv
+        $VIPassword = az keyvault secret show -n "vcenter-password-v2" --vault-name "${{ env.KEYVAULT }}" --query value -o tsv
```

Comment:
Can we save these variables as environment variables to not read in every step?

Reply:
@maxim-lobanov no, they will be revealed during step initialization as plain text

Reply:
Why can't we mark them as secret?
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-masking-an-environment-variable